docs/dev_guides/settings_validation.md

```diff
@@ -0,0 +1,142 @@
+
+Detailed information on the arguments that can be suppled is available within the [json schema documentation](https://json-schema.org/learn/getting-started-step-by-step).
+
+<hr>
+
+## Settings Validator
+
+In addition to the Settings Schema Validation, we have implemented checks to evaluate the validity of a user's settings in the context of the provided input dataframe(s) and linker type.
+
+??? note "The current validation checks performed include:"
+    * Verifying that the user's blocking rules and comparison levels have been [imported from the correct library](https://github.com/moj-analytical-services/splink/pull/1579) and contain sufficient information for use in a Splink model.
+    * [Performing column lookups]((https://github.com/moj-analytical-services/splink/blob/master/splink/settings_validation/column_lookups.py)), confirming that columns specified in the user's settings dictionary exist within **all** of the user's input dataframes.
+    * Conducting various miscellaneous checks that generate more informative error messages for the user, should they be using Splink in an unintended manner.
+
+All components related to our settings checks are currently located in the [settings validation](https://github.com/moj-analytical-services/splink/tree/32e66db1c8c0bed54682daf9a6fea8ef4ed79ab4/splink/settings_validation) within Splink.
+
+This folder is comprised of three scripts, each of which inspects the settings dictionary at different stages of its journey:
+
+* [valid_types.py](https://github.com/moj-analytical-services/splink/blob/32e66db1c8c0bed54682daf9a6fea8ef4ed79ab4/splink/settings_validation/valid_types.py) - This script includes various miscellaneous checks for comparison levels, blocking rules, and linker objects. These checks are primarily performed within settings.py.
+* [settings_validator.py](https://github.com/moj-analytical-services/splink/blob/32e66db1c8c0bed54682daf9a6fea8ef4ed79ab4/splink/settings_validation/settings_validator.py) - This script includes the core `SettingsValidator` class and contains a series of methods that retrieve information on fields within the user's settings dictionary that contain information on columns to be used in training and prediction. Additionally, it provides supplementary cleaning functions to assist in the removal of quotes, prefixes, and suffixes that may be present in a given column name.
+* [column_lookups.py](https://github.com/moj-analytical-services/splink/blob/32e66db1c8c0bed54682daf9a6fea8ef4ed79ab4/splink/settings_validation/column_lookups.py) - This script contains helper functions that generate a series of log strings outlining invalid columns identified within your settings dictionary. It primarily consists of methods that run validation checks on either raw SQL or input columns and assesses their presence in **all** dataframes supplied by the user.
+
+<hr>
```

Comment:
I would be tempted to split this up into 2 pages at this stage. One to introduce the validator, what it does and how it is structured. Then a second one focussing how to extend the functionality.